app/bundles/LeadBundle/Form/Type/EntityFieldsBuildFormTrait.php

```diff
@@ -43,6 +43,18 @@ private function getFormFields(FormBuilderInterface $builder, array $options, $o
         }
         $mapped = !$isObject;
 
+        $isUpdateOwnerLeadOrCompanyAction = false;
+        $traces                           = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 5);
+        foreach ($traces as $trace) {
+            if (isset($trace['class']) && (
+                'Mautic\LeadBundle\Form\Type\UpdateLeadActionType' === $trace['class']
+                || 'Mautic\LeadBundle\Form\Type\UpdateCompanyActionType' === $trace['class']
+            )) {
+                $isUpdateOwnerLeadOrCompanyAction = true;
+                break;
```

Comment:
This is smelly. Can this flag be for example passed through the `$options` instead?